internal/server/server.go

```diff
@@ -38,13 +39,26 @@ const (
 	Finalizer = "serverFinalizer.stormforge.io"
 )
 
+var (
+	// nameRegexp is used to validate the experiment labels that are required to be names.
+	nameRegexp = regexp.MustCompile(`^[a-z\d](?:[-a-z\d]{0,61}[a-z\d])?$`)
```

Comment:
should we have this in optimize-go? 
I don't think it's a blocker for this pr.

Reply:
[Yes](https://github.com/thestormforge/optimize-go/commit/6ff58273bd24d70178fb9cee874819b5f1d19158) ;)